iree/turbine/kernel/compiler/host_codegen.py

```diff
@@ -84,18 +89,40 @@ def isolated_test_call(
         ftype = FunctionType.get(input_tensors, output_tensors)
         func_op = func_d.FuncOp(func_name, ftype)
         actual_loc = FileLineColInfo.capture_current_location().to_mlir()
+        scalar_bindings = sig.scalar_bindings
         arg_locs = [
             (Location.name(b.name, actual_loc) if b.name is not None else actual_loc)
             for b in sig.kernel_buffer_bindings
-            + sig.scalar_bindings
+            + scalar_bindings
             + sig.dynamic_dim_bindings
         ]
         entry_block = func_op.add_entry_block(arg_locs)
-        offset = len(sig.kernel_buffer_bindings)
-        dynamic_argument_map = {
-            k: v for k, v in zip(dynamic_symbols, entry_block.arguments[offset:])
-        }
+        scalars_offset = len(sig.kernel_buffer_bindings)
+        scalars_count = len(scalar_bindings)
+        dynamic_offset = scalars_offset + scalars_count
+
+        def to_index(v: Value) -> Value:
```

Comment:
Can you pull this nested function out? breaks the flow when reading the code. Also might I suggest renaming to maybe_cast_to_index ? Might be worthwhile to add this to a utils file since it seems handy for other tasks as well.

Reply:
Moved func outside for now. I don't think we have a good common codegen utils module yet.

Reply:
thanks